sdk/servicebus/service-bus/test/perf/service-bus/receive.ts

```diff
@@ -0,0 +1,138 @@
+/*
+# Overview
+Measures the maximum throughput of `receiver.receive()` in package `@azure/service-bus`.
+
+# Instructions
+1. Create a Service Bus namespace with `Tier=Premium` and `Messaging Units=4`.  It is recommended to use the largest possible namespace to allow maximum client throughput.
+2. Create a queue inside the namespace.
+3. Set env vars `SERVICE_BUS_CONNECTION_STRING` and `SERVICE_BUS_QUEUE_NAME`.
+4. This test presumes that there are messages in the queue.
+4. `ts-node receive.ts [totalMessages]`
+5. Example: `ts-node receive.ts 1000000`
+ */
+
+import { ServiceBusClient, ReceiveMode, OnError, OnMessage, Receiver } from "../../../src";
+import delay from "delay";
+import moment from "moment";
+
+const _start = moment();
+
+let _messages = 0;
+
+/**
+ * Maximum wait duration for the expected event to happen = `10000 ms`(default value is 10 seconds)(= maxWaitTimeInMilliseconds)
+ * Keep checking whether the predicate is true after every `1000 ms`(default value is 1 second) (= delayBetweenRetriesInMilliseconds)
+ */
+async function checkWithTimeout(
```

Comment:
If we close the ServiceBusClient after the `writeResultsPromise` promise resolves, we won't need this, right?

Reply:
That's a good idea
But in that case, we would deviate from the format of the samples where have the `main()` module independent of the actual test. Is that okay ?

Reply:
**Update -**
- Removed `checkWithTimeout` 
- To close the receiver, client and namespace, added an if check inside the OnMessage handler.
https://github.com/Azure/azure-sdk-for-js/blob/527176e7d6860661d1d22e0d44f36c5db079a503/sdk/servicebus/service-bus/test/perf/service-bus/receive.ts#L47-L54

Reply:
> But in that case, we would deviate from the format of the samples where have the main() module independent of the actual test. Is that okay ?

I don't feel too strongly about that.

How does the .net test behave? At what point is the streaming stopped?